### DISCUSSION TITLE
Issues with menu bar icon & dragging windows.

### DISCUSSION THREAD

**Discussion by theorist1** [rxhanson/RectanglePro-Community#419] [Q&A]
Hello.  I'm trying out Rectange Pro in a 2019 i9 iMac running Monterery 12.7.2. Three questions:

1) When I click on the icon in the menu bar, I expect it to display the dropdown with the various shortcuts. Instead, clicking just changes the icon from this:
<img width="58" alt="image" src="https://github.com/rxhanson/RectanglePro-Community/assets/9023016/98d41f5a-aeca-4168-bbc5-270ed4fc05c4">

to this (i.e., it just highlights the space around the icon):
<img width="88" alt="image" src="https://github.com/rxhanson/RectanglePro-Community/assets/9023016/a5467a47-dc58-478c-8727-15e0dbbb75b3">

I have to click on it several times to get the dropdown, i.e., to get this:

<img width="273" alt="image" src="https://github.com/rxhanson/RectanglePro-Community/assets/9023016/a019147a-9103-4a65-b264-cedc6c96ee36">

Is there any way I can get Rectangle to display the dropdown with just a single click?  That's how all the other icons in my menu bar work.

2) Suppose I have a window that is half-screen, and I drag it to the center.   The behavior I expect is that the window will remain the same size, but be repositioned.  But when running Rectange, the window sometimes shrinks, and sometimes expands to fill the screen.  How do I stop Rectange from changing the size of windows as a result of dragging them?

Curiously, this behavior depends on what is being displayed.  For instance, with Safari windows, the shrinking behavior is seen when I am viewing nytimes.com without the Bookmarks sidebar.  If I display the sidebar, the window will not shrink. Alternately, when I display this github page, I don't see the shrinking behavior either with or without the Bookmarks sidebar.    

I've never noticed any of the behavior I'm describing here when using "SizeUp!", which is similar in function to Rectange.
 
3)  Is there a user manual for this?

**Comment by rxhanson** [comment]
Thanks for trying Rectangle Pro.

1. The only time I see this kind of behavior is when using multiple displays and clicking on the menu bar item on a display that is not active (containing the front/key window). Is this where you also see it?
2. There's a checkbox in the Snap Areas tab of the settings window: "Restore window size when unsnapped". I think that's what you're looking for.
3. There is currently no documentation for the app aside from this repository and the info buttons within the app. I created this video about the window throw: https://www.youtube.com/watch?v=Vo09H-ln7BQ

**Comment by theorist1** [comment]
Thanks for your quick reply!

1) That is indeed what's happening.  I run three displays, but typically prefer to use the center one to check menu bar icons.  Thus it's exactly when one of my other two displays is active that I see this behavior.  The workaround is to first click on the center display to make it active, and then click on Rectangle's icon.    

However, I have 24 other menu bar icons, and none of them exhibit this behavior.   It doesn't matter which display is active—when I click on them, I get their dropdowns.  This is true for both built-in Apple functions (e.g., Time Machine, the wireless networks, and the Keyboard Viewer), as well as for third-party apps (e.g., f.lux, Carbon Copy Cloner, Keyboard Maestro, and SizeUp!).  Rectangle's icon is the only one that requires the window be active in order to respond.  

I'm curious why this is but, at the same time, once I've established the preferences I won't need to check the dropdown often, so this shouldn't be a practical issue.

2) Actually, I already had **Restore window size when unsnapped** checked.  The issue was that I had to *uncheck* **Snap windows by dragging**.  But still, your suggestion did point me in the right direction.

These minor wrinkles aside, overall the program seems to work beautifully.

**Comment by theorist1** [comment]
Wow--I think that's the first time I've mentioned an issue and saw a fix in the next update!

I just installed it, and can confirm your fix for #1 works on my system.
